mesonbuild/interpreter/interpreter.py

```diff
@@ -2467,33 +2467,35 @@ def func_install_data(self, node: mparser.BaseNode,
                     '"rename" and "sources" argument lists must be the same length if "rename" is given. '
                     f'Rename has {len(rename)} elements and sources has {len(sources)}.')
 
+        install_dir = kwargs['install_dir']
+        if not install_dir:
+            subdir = self.active_projectname
+            install_dir = P_OBJ.OptionString(os.path.join(self.environment.get_datadir(), subdir), os.path.join('{datadir}', subdir))
```

Comment:
```python
if self.is_subproject():
    FeatureNew.single_use('install_data without install_dir inside of a subproject', '1.2.0', self.subproject,
                          'This was broken and would install to the project name of the parent project instead',
                          node)

Reply:
s/1.2.0/1.3.0. I thought there was a FeatureBroken or something of the sort

Reply:
FeatureBroken is for stuff that you want to print unconditional deprecations and say "never ever ever use this".

Reply:
This implies that we cannot have the fix in a potential 1.2.1 release. Can we?

Reply:
Good point. I will leave it to Eli to determine that.